xunit/xunit
# Windows Phone App 8.1 and Windows Store App support

This PR contains WPA 8.1 and Win8 support for xUnit.
## Notes:

Win8 was fairly simple -- just needed .net 4.5 versions w/o appdomains.

WPA 8.1 was very complicated. WPA 8.1 needs device-specific files (so it needs to be compiled for the Wpa 8.1 profile). That meant using the "new reflection" API for Execution (which is 4.5 only). It also meant that Binary Serialization had to be worked around.

On top of that, the VS Runner needs to load a desktop version of the Executor dll from the output directory, but a Wpa81 version needs to be copied to the device.

For WPA 8.1, and only on the WPA 8.1 app project type, it does need a Json.net reference. Json.net was the only json lib that can round-trip based on full .net type information. This change was done in IFDEF's and has zero impact on any other xUnit platforms.
## To Use
### Windows Store App
1.  Create a new Windows Store Unit Test Project.
2.  Remove the MSTestFramework reference
3.  Add the xUnit NuGet reference.
4.  Make sure you have the xUnit VS Runner installed. There's no other way currently to run these tests
### Windows Phone App 8.1
1.  Create a new Windows Phone App Unit Test project
2.  Remove the MSTestFamework Reference.
3.  Add the xUnit NuGet reference
4.  Make sure you have the xUnit VS Runner installed. There's no other way currently to run these tests
## Known issues

Currently Any CPU or x86/x64 must be the selected type for your Test App. ARM libraries are currently failing to be discovered.
